plugins/sonar-xoo-plugin/src/main/java/org/sonar/xoo/test/TestExecutionSensor.java

```diff
@@ -0,0 +1,119 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.xoo.test;
+
+import com.google.common.base.Splitter;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.sonar.api.batch.DependedUpon;
+import org.sonar.api.batch.fs.FilePredicates;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.fs.InputFile.Type;
+import org.sonar.api.batch.sensor.Sensor;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.batch.sensor.SensorDescriptor;
+import org.sonar.api.component.ResourcePerspectives;
+import org.sonar.api.test.MutableTestCase;
+import org.sonar.api.test.MutableTestPlan;
+import org.sonar.api.test.TestCase.Status;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.xoo.Xoo;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Parse files *.xoo.test
+ */
+@DependedUpon("test-exec")
+public class TestExecutionSensor implements Sensor {
+  private static final Logger LOG = Loggers.get(TestExecutionSensor.class);
+
+  private static final String TEST_EXTENSION = ".test";
+
+  private final FileSystem fs;
+  private final ResourcePerspectives perspectives;
+
+  public TestExecutionSensor(FileSystem fileSystem, ResourcePerspectives perspectives) {
+    this.fs = fileSystem;
+    this.perspectives = perspectives;
+  }
+
+  private void processTestFile(InputFile inputFile, SensorContext context) {
+    File testExecutionFile = new File(inputFile.file().getParentFile(), inputFile.file().getName() + TEST_EXTENSION);
+    if (testExecutionFile.exists()) {
+      LOG.debug("Processing " + testExecutionFile.getAbsolutePath());
+      try {
+        List<String> lines = FileUtils.readLines(testExecutionFile, fs.encoding().name());
+        int lineNumber = 0;
+        MutableTestPlan testPlan = perspectives.as(MutableTestPlan.class, inputFile);
+        for (String line : lines) {
+          lineNumber++;
+          if (StringUtils.isBlank(line)) {
+            continue;
+          }
+          if (line.startsWith("#")) {
+            continue;
+          }
+          try {
+            Iterator<String> split = Splitter.on(":").split(line).iterator();
+            String name = split.next();
+            String durationStr = split.next();
+            Long duration = StringUtils.isNotBlank(durationStr) ? Long.parseLong(durationStr) : null;
+            String msg = split.next();
+            String stack = split.next();
+            String status = split.next();
+            String type = split.next();
+            MutableTestCase testCase = testPlan.addTestCase(name);
```

Comment:
:new: ![BLOCKER](https://raw.githubusercontent.com/henryju/image-hosting/master/blocker.png)NullPointerException might be thrown as 'testPlan' is nullable here[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2259)